client/image_push.go

```diff
@@ -36,6 +38,16 @@ func (cli *Client) ImagePush(ctx context.Context, image string, options image.Pu
 		}
 	}
 
+	if options.Platform != nil {
```

Comment:
Perhaps we need a version-error here; similar to https://github.com/moby/moby/blob/cd3804655a252117742a2987df15630a61ce334e/client/image_build.go#L126-L129

Reply:
Good catch, done!